Inconsistent native and service KCL api `ParseFile` behaviour

## Bug Report

Native KCL api `ParseFile` method returns rust panicked error.

### 1. Minimal reproduce step (Required)

Test for reproduce: https://github.com/bozaro/kcl-parse-file-issue/blob/master/kcl_parse_file_issue_test.go

1. Clone repository:
```
git clone https://github.com/bozaro/kcl-parse-file-issue.git .
```
2. Run test:
```
go test -v .
```

### 2. What did you expect to see? (Required)
I expect to see same result for native and service tests.

### 3. What did you see instead (Required)
Native client returns rust panicked error with kcl parse non-empty errors case:
```
=== RUN   Test
=== RUN   Test/pygrpc
=== RUN   Test/pygrpc/simple
=== RUN   Test/pygrpc/pkgpath_not_found
=== RUN   Test/native
=== RUN   Test/native/simple
=== RUN   Test/native/pkgpath_not_found
thread '<unnamed>' panicked at api/src/service/capi.rs:154:5:
called `Result::unwrap()` on an `Err` value: NulError(7238, [10, 139, 55,..., 111, 39, 18, 0])
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
    kcl_parse_file_issue_test.go:64: 
        	Error Trace:	/home/bozaro/xxx/xxa/kcl_parse_file_issue_test.go:64
        	Error:      	ParseFile returns error: %s
        	Test:       	Test/native/pkgpath_not_found
        	Messages:   	called `Result::unwrap()` on an `Err` value: NulError(7238, [10, 139, 55, 123, 34, 102, 105, 108, 10
--- FAIL: Test (0.01s)
    --- PASS: Test/pygrpc (0.00s)
        --- PASS: Test/pygrpc/simple (0.00s)
        --- PASS: Test/pygrpc/pkgpath_not_found (0.00s)
    --- FAIL: Test/native (0.00s)
        --- PASS: Test/native/simple (0.00s)
        --- FAIL: Test/native/pkgpath_not_found (0.00s)
FAIL
FAIL	github.com/bozaro/kcl-parse-file-issue	0.014s
FAIL
```

### 4. What is your KCL components version? (Required)
```
kcl-lang.io/kcl-go v0.8.1
```

--- a/pkg/native/cgo.go
+++ b/pkg/native/cgo.go
@@ -23,10 +23,10 @@ void kclvm_service_free_string(void *f,const char * res) {
 	free_string = (void (*)(const char *))f;
 	return free_string(res);
 }
-const char* kclvm_service_call(void *f,kclvm_service* c,const char * method,const char * args){
-	const char* (*service_call)(kclvm_service*,const char *,const char *);
-	service_call = (const char* (*)(kclvm_service*,const char *,const char *))f;
-	return service_call(c,method,args);
+const char* kclvm_service_call_with_length(void *f,kclvm_service* c,const char * method,const char * args,size_t * result_len){
+	const char* (*service_call_with_length)(kclvm_service*,const char *,const char *,size_t *);
+	service_call_with_length = (const char* (*)(kclvm_service*,const char *,const char *,size_t *))f;
+	return service_call_with_length(c,method,args,result_len);
 }
 */
 import "C"
@@ -75,8 +75,8 @@ func KclvmServiceFreeString(str *C.char) {
 
 // KclvmServiceCall calls kclvm service by c api
 // args should be serialized as protobuf byte stream
-func KclvmServiceCall(serv *C.kclvm_service, method *C.char, args *C.char) *C.char {
-	const fnName = "kclvm_service_call"
+func KclvmServiceCall(serv *C.kclvm_service, method *C.char, args *C.char) (*C.char, C.size_t) {
+	const fnName = "kclvm_service_call_with_length"
 
 	serviceCall, err := lib.GetSymbolPointer(fnName)
 
@@ -84,5 +84,7 @@ func KclvmServiceCall(serv *C.kclvm_service, method *C.char, args *C.char) *C.ch
 		panic(err)
 	}
 
-	return C.kclvm_service_call(serviceCall, serv, method, args)
+	var size C.size_t
+	buf := C.kclvm_service_call_with_length(serviceCall, serv, method, args, &size)
+	return buf, size
 }

--- a/pkg/native/client.go
+++ b/pkg/native/client.go
@@ -10,6 +10,7 @@ typedef struct kclvm_service kclvm_service;
 */
 import "C"
 import (
+	"bytes"
 	"errors"
 	"runtime"
 	"strings"
@@ -84,20 +85,20 @@ func cApiCall[I interface {
 
 	defer C.free(unsafe.Pointer(cIn))
 
-	cOut := KclvmServiceCall(c.client, cCallName, cIn)
+	cOut, cOutSize := KclvmServiceCall(c.client, cCallName, cIn)
 
 	defer KclvmServiceFreeString(cOut)
 
-	msg := C.GoString(cOut)
+	msg := C.GoBytes(unsafe.Pointer(cOut), C.int(cOutSize))
 
-	if strings.HasPrefix(msg, "ERROR:") {
-		return nil, errors.New(strings.TrimPrefix(msg, "ERROR:"))
+	if bytes.HasPrefix(msg, []byte("ERROR:")) {
+		return nil, errors.New(strings.TrimPrefix(string(msg), "ERROR:"))
 	}
 
 	var out O = new(TO)
-	err = proto.Unmarshal([]byte(msg), out)
+	err = proto.Unmarshal(msg, out)
 	if err != nil {
-		return nil, errors.New(msg)
+		return nil, errors.New(string(msg))
 	}
 
 	return out, nil

